tiledb/sm/cpp_api/array.h

```diff
@@ -1505,6 +1511,8 @@ class Array {
   /** The array schema. */
   ArraySchema schema_;
 
+  bool dontclose_ = false;
```

Comment:
Could you explain why this is needed? I don't want to introduce this mode into the API. Also, please separate words in a variable name with `_`.

Reply:
Just saw this, yes let's stick to snake case as I mentioned above.

Reply:
There is a cppapi constructor that takes a tiledb_array_t and an 'own' parameter.
There is arguably a bug in the implementation of 'own' within the class, where it close()s that array even though it does not own it.  But 'fix'ing the behaviour of 'own' to not close() would be a breaking change for any currently existing code that is dependent on that behaviour.  (The close() behaviour is documented for every other constructor except that one, but users of that constructor are subject to the current behaviour irrespective of documentation.)
The compensatory 'dont_close" parameter was added for use in (subsequent story) cloned/modified SubarrayPartioner tests whose infrastructure used a pre-existing tiledb_array_t where that infrastructure expects it to remain open.  It also supports providing what I believe is more correct behaviour in the class for similar situations for !own'd tiledb_array_t entities.